### DISCUSSION TITLE
SLURM Headnode nompi

### DISCUSSION THREAD

**Discussion by jwarners772** [neperfepx/FEPX#107] [Q&A] [answered]
It seems it's required to install openmpi and I get the error output when running cmake.

-- Could NOT find MPI_Fortran (missing: MPI_Fortran_LIB_NAMES MPI_Fortran_F77_HEADER_DIR MPI_Fortran_MODULE_DIR MPI_Fortran_WORKS)
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_Fortran_FOUND)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindMPI.cmake:1742 (find_package_handle_standard_args)
  CMakeLists.txt:12 (find_package)

-- Configuring incomplete, errors occurred!
See also "/opt/crystalplasticity/FEPX/src/CMakeFiles/CMakeOutput.log".
See also "/opt/crystalplasticity/FEPX/src/CMakeFiles/CMakeError.log".

Typically we install the software onto our head node and then pass it to the cluster which does have openmpi. Is there anyway for me to call a parameter or alter the cmake to allow the install to complete without openmpi's installation?

Thanks!

**Comment by mkasemer** [comment, marked as answer]
@jwarners772 : The errors indicate that CMake cannot find MPI Fortran. Usually, using OpenMPI, it is looking for the `mpif90` binary. If you know this to exist and know where it is located, you can manually set this using CMake options.

In any event, this is a server configuration / CMake issue.

**Reply by jwarners772** [reply]
That did it thank you! 

cmake -DMPI_Fortran_COMPILER=/usr/bin/mpif90 .
